src/hotspot/share/ci/ciMethodData.cpp

```diff
@@ -83,7 +79,7 @@ class PrepareExtraDataClosure : public CleanExtraDataClosure {
 
   bool is_live(Method* m) {
     Klass* holder = m->method_holder();
-    if (!has_cld(holder) || !holder->is_loader_alive() || !is_klass_loaded(holder)) {
+    if (holder == nullptr || !holder->is_loader_present_and_alive() || !is_klass_loaded(holder)) {
```

Comment:
Why do we need these checks on CI side?

Reply:
Wasn't it you who put it there? :)

Reply:
Nevermind, it turns out to be just a refactoring:`holder == nullptr || !holder->is_loader_present_and_alive()` is equivalent to the original check `!has_cld(holder) || !holder->is_loader_alive()`.